sentry-ruby/spec/sentry_spec.rb

```diff
@@ -784,4 +784,141 @@
     end
   end
 
+  describe ".close" do
+    context "when closing initialized SDK" do
+      it "not initialized?" do
+        expect(described_class.initialized?).to eq(true)
+        described_class.close
+        expect(described_class.initialized?).to eq(false)
+      end
+
+      it "removes main hub" do
+        expect(described_class.get_main_hub).to be_a(Sentry::Hub)
+        described_class.close
+        expect(described_class.get_main_hub).to eq(nil)
+      end
+
+      it "removes thread local" do
+        expect(Thread.current.thread_variable_get(described_class::THREAD_LOCAL)).to be_a(Sentry::Hub)
+        described_class.close
+        expect(Thread.current.thread_variable_get(described_class::THREAD_LOCAL)).to eq(nil)
+
+      end
+
+      it "calls background worker shutdown" do
+        expect(described_class.background_worker).to receive(:shutdown)
+        described_class.close
+        expect(described_class.background_worker).to eq(nil)
+      end
+
+      it "kills session flusher" do
+        expect(described_class.session_flusher).to receive(:kill)
+        described_class.close
+        expect(described_class.session_flusher).to eq(nil)
+      end
+
+      it "disables Tracepoint" do
+        perform_basic_setup do |config|
+          config.capture_exception_frame_locals = true
+        end
+
+        expect(described_class.exception_locals_tp).to receive(:disable)
```

Comment:
This blocks the TP to actually receive `disable` so it'll be left enabled and break other tests.

Reply:
ahh thanks, turns out I still don't get rspec

Reply:
You can add `.and_call_original` to let it still perform the call.